src/browser_action/on-each-interaction.js

```diff
@@ -60,16 +60,11 @@ export function onEachInteraction(callback) {
     for (const interaction of Object.values(interactions)) {
       const entry = interaction.reduce((prev, curr) => prev.duration >= curr.duration ? prev : curr);
       const value = entry.duration;
-      const interactionId = entry.interactionId;
 
       // Filter down LoAFs to ones that intersected any event startTime and any processingEnd
       const longAnimationFrameEntries = getIntersectingLoAFs(entry.startTime, entry.startTime + entry.value)
 
-      let firstEntryWithTarget = interaction[0].target;
-      firstEntryWithTarget = null;
-      if (!firstEntryWithTarget) {
-        firstEntryWithTarget = Object.values(interactions).flat().find(entry => entry.interactionId === interactionId && entry.target)?.target;
-      }
+      const firstEntryWithTarget = interaction.find(entry => entry.target)?.target;
```

Comment:
The point of setting it, was to save on the (potentially costly) `find` lookup and only do that when needed.

Also, with buffered entries, we might have a whole history if different interactions so only want to look at the interactions with the same interactionId to get the target as otherwise we might incorrectly report earlier targets for later interactions (if the performance observer callback is invoked with that all at once).

```suggestion
      let firstEntryWithTarget = interaction[0].target;
      if (!firstEntryWithTarget) {
        firstEntryWithTarget = Object.values(interactions).flat().find(entry => entry.interactionId === interactionId && entry.target)?.target;
      }
```

Reply:
But the `interactions` object is already grouped by `interactionId` so we shouldn't need to filter it down again. We can just look at the array of entries for the `interactionId` we are interested in. Or am I misunderstanding something?

Also `find` exits early if the first item in the array satisfies it's predicate, so we wouldn't loop through the entire array if we don't need to.

Reply:
Oh good points!

OK think you're right. Let's merge as is.